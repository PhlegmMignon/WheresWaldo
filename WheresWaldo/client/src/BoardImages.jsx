import convention from "./assets/nMpQXwq.jpg";
import convention_kon from "./assets/kon.webp";
import convention_luffy from "./assets/luffy.png";
import convention_zim from "./assets/Zim.yelling.svg";

import stage2 from "./assets/stage3.png";
import stage2_fish from "./assets/fish.png";
import stage2_cannon from "./assets/cannon.png";
import stage2_egg from "./assets/egg.png";

export default function BoardImages() {
  const getImages = () => {
    return [
      {
        name: "convention",
        src: convention,
        id: 0,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
      },
      {
        name: "stage 2",
        src: stage2,
        id: 1,
        characters: [
          { name: "fish", src: stage2_fish, id: 0 },
          { name: "cannon", src: stage2_cannon, id: 1 },
          { name: "egg", src: stage2_egg, id: 2 },
        ],
      },
    ];
  };

  return { getImages };
}
