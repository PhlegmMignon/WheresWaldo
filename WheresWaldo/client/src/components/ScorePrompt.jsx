import { useState } from "react";

export default function ScorePrompt({
  setIsHighscore,
  time,
  gameImage,
  getScores,
}) {
  const [name, setName] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();

    const details = { name: name, score: time, map: gameImage.name };

    fetch("http://localhost:3000/scores", {
      method: "POST",
      mode: "cors",
      credentials: "include",
      headers: {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "http://localhost:5173",
      },
      body: JSON.stringify(details),
    })
      .then((res) => {
        return res.json();
      })
      .then((res) => {
        res.success ? setIsHighscore(false) : console.log("Err on score post");
        return;
      })
      .catch((err) => console.log("Err on score post" + err));

    //Updates scores with new highscore
    getScores(gameImage.name);
  };

  const updateHighscore = () => {
    getScores(gameImage.name);
    setIsHighscore(false);
  };

  return (
    <div
      id="scorePrompt"
      className="absolute inset-1/2 flex flex-col gap-8 p-4 h-56 w-56 rounded-lg"
    >
      <div className="text-center text-lg">A new highscore!</div>
      <form
        onSubmit={handleSubmit}
        method="post"
        action="/scores"
        className="flex flex-col gap-2 "
      >
        <label htmlFor="name">Name</label>
        <input
          required
          onChange={(e) => setName(e.target.value)}
          name="name"
          type="text"
          className="w-full"
        />
        <div className="flex gap-14 mt-2.5 ml-7">
          <button type="submit">Submit</button>
          <button onClick={updateHighscore}>Skip</button>
        </div>
      </form>
    </div>
  );
}
