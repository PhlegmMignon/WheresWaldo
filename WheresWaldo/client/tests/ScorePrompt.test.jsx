import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";
import ScorePrompt from "../src/components/ScorePrompt.jsx";
import BoardImages from "../src/BoardImages.jsx";

describe("ScorePrompt unit tests", () => {
  it("Renders stage select", async () => {
    const { container } = render(<ScorePrompt />);

    expect(container).toMatchSnapshot();
  });
});

describe("ScorePrompt integration tests", () => {
  //This test is simple but time consuming to make, so I'm skipping it.
  //I should've pulled fetch requests into their own functions so it would be more reusable and easier to mock.
  // it("Submitting form calls handleSubmit", async () => {
  //   vi.fn().mockResolvedValueOnce
  //   const user = userEvent.setup();
  //   const fakeGetScores = vi.fn();

  //   render(<ScorePrompt getScores={fakeGetScores} />);

  //   const btn = screen.getByRole("button", { name: "Submit" });
  //   await user.click(btn);

  //   expect(fetch.requests().length).toEqual(1);
  // });

  it("Skipping form calls updateHighScore", async () => {
    const user = userEvent.setup();
    const fakeGetScores = vi.fn();
    const fakeSetHighScore = vi.fn();

    let gameImage = BoardImages().getImages();

    render(
      <ScorePrompt
        getScores={fakeGetScores}
        gameImage={gameImage[0].name}
        setIsHighscore={fakeSetHighScore}
      />
    );

    const btn = screen.getByRole("button", { name: "Skip" });
    await user.click(btn);

    expect(fakeGetScores).toHaveBeenCalled();
    expect(fakeSetHighScore).toHaveBeenCalled();
  });
});
