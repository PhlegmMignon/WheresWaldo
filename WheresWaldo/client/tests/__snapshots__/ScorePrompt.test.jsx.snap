// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScorePrompt unit tests > Renders stage select 1`] = `
<div>
  <div
    class="absolute inset-1/2 flex flex-col gap-8 p-4 h-56 w-56 rounded-lg"
    id="scorePrompt"
  >
    <div
      class="text-center text-lg"
    >
      A new highscore!
    </div>
    <form
      action="/scores"
      class="flex flex-col gap-2 "
      method="post"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        class="w-full"
        name="name"
        required=""
        type="text"
      />
      <div
        class="flex gap-14 mt-2.5 ml-7"
      >
        <button
          type="submit"
        >
          Submit
        </button>
        <button>
          Skip
        </button>
      </div>
    </form>
  </div>
</div>
`;
